src/webgpu/api/validation/render_pass/render_pass_descriptor.spec.ts

```diff
@@ -848,6 +848,47 @@ g.test('resolveTarget,format_supports_resolve')
     });
   });
 
+g.test('timestampWrites,query_set_type')
+  .desc(
+    `
+  Test that all entries of the timestampWrites must have type 'timestamp'. If all query types are
+  not 'timestamp', a validation error should be generated.
+  `
+  )
+  .params(u =>
+    u //
+      .combine('queryTypeA', kQueryTypes)
+      .combine('queryTypeB', kQueryTypes)
+  )
+  .beforeAllSubcases(t => {
+    t.selectDeviceOrSkipTestCase(['timestamp-query']);
+  })
+  .fn(async t => {
+    const { queryTypeA, queryTypeB } = t.params;
+
+    const timestampWriteA = {
+      querySet: t.device.createQuerySet({ type: queryTypeA, count: 1 }),
+      queryIndex: 0,
+      location: 'beginning' as GPURenderPassTimestampLocation,
+    };
+
+    const timestampWriteB = {
+      querySet: t.device.createQuerySet({ type: queryTypeA, count: 1 }),
+      queryIndex: 0,
+      location: 'beginning' as GPURenderPassTimestampLocation,
```

Comment:
The locations need to be different, or it's invalid (as tested by the test just below this one): https://gpuweb.github.io/gpuweb/#abstract-opdef-gpucomputepassdescriptor-valid-usage

(I don't know if that test passes in chromium or not)

Reply:
Ah, right. But, Chromium is passing the test even if the locations are equal. Fixed.